bpf/lib/nodeport.h

```diff
@@ -2106,10 +2105,8 @@ nodeport_extract_dsr_v4(struct __ctx_buff *ctx,
 #endif
 
 	/* SYN for a new connection that's not / no longer DSR.
-	 * If it's reopened, avoid sending subsequent traffic down the DSR path.
+	 * If it's reopened, the CT entry will be re-created with a cleared dsr flag.
 	 */
-	if (tuple->nexthdr == IPPROTO_TCP)
-		ct_update_dsr(get_ct_map4(&tmp), &tmp, false);
 
```

Comment:
I'm not 100% on this. Right now it covers not just `CT_REOPENED`, but also the case where we receive multiple SYNs (by a oddly-behaving client :)). The 2nd/3rd/... SYN would get matched as `CT_ESTABLISHED`, no? So we still want to update the DSR-state accordingly.

Reply:
This was not done for `CT_ESTABLISHED`, so this change is correct IMO. Also, I don't know if it is possible for dsr to be `true` in the case you refer to (re-transmitted `SYN`s).

Reply:
You are right, I had this mixed up. Reverted this change (also for v6).